### PR TITLE
Remove vsce as dev dependency

### PR DESCRIPTION
**Pull request by MetRonnie**
GH actions deploy workflow should work without vsce being a dev dependency; have changed it to install vsce globally in the workflow enviroment.

Plus some documentation changes.

### CODE REVIEW

File: .github/workflows/deploy.yml [hunk 1]
```diff
@@ -24,13 +24,13 @@ jobs:
...
-    - name: Install dependencies
-      run: npm install
+    - name: Install vsce
+      run: npm install -g vsce
 
     - name: Publish to VSCode Marketplace
       env:
         PAT: ${{ secrets.MARKETPLACE_PAT }}
-      run: npx vsce publish -p "$PAT"
+      run: vsce publish -p "$PAT"
```
Comment (wxtim) [inline .github/workflows/deploy.yml:33]: what is `$PAT`?

File: README.md [hunk 2]
```diff
@@ -20,21 +20,25 @@ Check [here](https://github.com/cylc/cylc-flow/issues/2752) for info on supporte
...
+This repo includes the [cylc/cylc-textmate-grammar](https://github.com/cylc/cylc-textmate-grammar) repo as a git submodule in the `/syntaxes/` directory. If you don't have experience with submodules, you should [read the docs](https://git-scm.com/book/en/v2/Git-Tools-Submodules) first.
 
 The cylc-textmate-grammar repo contains a JSON TextMate grammar file which is used by VSCode for syntax highlighting. Read the [VSCode syntax highlight guide](https://code.visualstudio.com/api/language-extensions/syntax-highlight-guide) for more information. **Note:** do not edit the JSON file when contributing; instead you should edit the JavaScript grammar file and build it, as explained in the [contributing](https://github.com/cylc/cylc-textmate-grammar#contributing) section of cylc-texmate-grammar.
 
-To install a development version of this extension, first find your VSCode extensions folder (should normally be `~/.vscode/extensions/`). Then:
```
Comment (wxtim) [inline README.md:29]: Why is this bit no longer rqd?

Comment (MetRonnie) [reply README.md:29]: It seems this is not the intended way to develop extensions, instead you're meant to clone to a different location and run a debugging session using F5: https://github.com/microsoft/vscode/issues/25159#issuecomment-296709441

Comment (wxtim) [reply README.md:29]: Doesn't surprise me - what I was doing was quick and dirty.

**Review by oliver-sanders — APPROVED** [review]

**Pull request merged**